### PR TITLE
add parents to scala 2 classes, erase scala 2 value classes

### LINKED ISSUE #119: Erasure of value class
Here is the test that fails:

```scala
testWithContext("scala.collection.ArrayOps") {
  val Predef = resolve(name"scala" / tname"Predef" / obj).asClass
  val genericArrayOps = Predef.getDecl(name"genericArrayOps").get
  assertIsSignedName(genericArrayOps.signedName, "genericArrayOps", "(1,java.lang.Object):java.lang.Object")
}
```

Notice that `ArrayOps[T]` is a value class of `Array[T]` and should be erased to `Object`.

### PR DESCRIPTION
**Pull request by bishabosha**
also standardise on paramRefss rather than paramSymss.

fixes #119

### CODE REVIEW

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala [hunk 1]
```diff
@@ -379,10 +389,14 @@ object Types {
...
-        case tpe: AppliedType => tpe.underlying.isRef(sym)
-        case tpe: ExprType    => tpe.underlying.isRef(sym)
-        case _                => false // todo: add ProxyType (need to fill in implementations of underlying)
+        case tpe: NamedType    => tpe.symbol == sym
+        case tpe: AppliedType  => tpe.underlying.isRef(sym)
+        case tpe: ExprType     => tpe.underlying.isRef(sym)
+        case tpe: TermParamRef => tpe.underlying.isRef(sym)
+        case tpe: TypeParamRef =>
+          val bounds = tpe.bounds
+          bounds.low.isRef(sym) || bounds.high.isRef(sym)
```
Comment (sjrd) [inline tasty-query/shared/src/main/scala/tastyquery/Types.scala:398]: Shouldn't this be an `&&`? Or only test `bounds.high`?

File: tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 2]
```diff
@@ -210,15 +214,26 @@ class PickleReader {
             .asInstanceOf[DeclaringSymbol]
             .getModuleDeclInternal(name.toTermName.stripObjectSuffix)
           module.foreach(m => m.asTerm.withDeclaredType(cls.typeRef))
+        val tpe = readSymType()
         val typeParams = atNoCache(infoRef)(readTypeParams())
+        if !isRefinementClass(cls) then
+          val parentTypes = tpe match
+            case TempPolyType(tparams, restpe) =>
+              assert(tparams.corresponds(typeParams)(_ eq _)) // should reuse the class type params
+              restpe.asInstanceOf[TempClassInfoType].parentTypes
```
Comment (sjrd) [inline tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:223]: Consider:
```suggestion
            case TempPolyType(tparams, restpe: TempClassInfoType) =>
              assert(tparams.corresponds(typeParams)(_ eq _)) // should reuse the class type params
              restpe.parentTypes
```

**Review by sjrd — CHANGES_REQUESTED** [review]

File: tasty-query/shared/src/main/scala/tastyquery/Types.scala (hunk 1 above)
Comment (bishabosha) [reply tasty-query/shared/src/main/scala/tastyquery/Types.scala]: ive changed to only check `high`

File: tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala (hunk 2 above)
Comment (bishabosha) [reply tasty-query/shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala]: done

**Review by sjrd — APPROVED** [review]

**Pull request merged**